assignments/prototype-refactor.js

```diff
@@ -7,3 +7,170 @@ Prototype Refactor
+  * destroy() // prototype method -> returns the string: 'Object was removed from the game.'
+*/
+    class GameObject {
+        constructor(charAtt){
+            this.createdAt = charAtt.createdAt;
+            this.dimensions = charAtt.dimensions;
+        }
+        destroy(){
+            return `${this.name} was removed from the game.`;  
+        }
+    }
+// function GameObject(charAtt) {
+    // this.createdAt = charAtt.createdAt;
+    // this.dimensions = charAtt.dimensions;
+//   }
+  
+//   GameObject.prototype.destroy = function () {
+//     return `${this.name} was removed from the game.`;
+//   }
+  
+
+  
+  /*
+    === CharacterStats ===
+    * healthPoints
+    * name
+    * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+    * should inherit destroy() from GameObject's prototype
+  */
+  
+  // function CharacterStats(charStats) {
+  //   GameObject.call(this, charStats)
+  //   this.healthPoints = charStats.healthPoints;
+  //   this.name = charStats.name;
+  // }
+  // CharacterStats.prototype = Object.create(GameObject.prototype);
+  // CharacterStats.prototype.takeDamage = function () {
+  //   return `${this.name} took damage.`
+  // }
+
+class CharacterStats extends GameObject{
+  constructor(charStats){
+   super(charStats)
+   this.healthPoints = charStats.healthPoints;
+   this.name = charStats.name;
+  }
+  takeDamage(){
+    return `${this.name} took damage.`
+  }
+}
+  /*
+    === Humanoid (Having an appearance or character resembling that of a human.) ===
+    * team
+    * weapons
+    * language
+    * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+    * should inherit destroy() from GameObject through CharacterStats
+    * should inherit takeDamage() from CharacterStats
+  */
+  
+  // function Humanoid(classAtt) {
+  //   CharacterStats.call(this, classAtt)
+  
+  //   this.team = classAtt.team;
+  //   this.weapons = classAtt.weapons;
+  //   this.language = classAtt.language;
+  // }
+  
+  // Humanoid.prototype = Object.create(CharacterStats.prototype);
+  
+  // Humanoid.prototype.greet = function () {
+  //   return `${this.name} offers a greeting in ${this.language}.`
+  // }
+
+  class Humanoid extends CharacterStats{
+    constructor(classAtt){
+      super(classAtt)
+      this.team = classAtt.team;
+      this.weapons = classAtt.weapons;
+      this.language = classAtt.language;
+    }
+    greet() {
+      return `${this.name} offers a greeting in ${this.language}.`
+    }
+  }
+  
+  /*
+   * Inheritance chain: GameObject -> CharacterStats -> Humanoid
+   * Instances of Humanoid should have all of the same properties as CharacterStats and GameObject.
+   * Instances of CharacterStats should have all of the same properties as GameObject.
+   */
+  
+  // Test you work by un-commenting these 3 objects and the list of console logs below:
+  
+  
+  const mage = new Humanoid({
```

Comment:
So now that you have spent time with this whole material over the weekend. What are we passing in as an argument inside of our `new Humanoid` function?